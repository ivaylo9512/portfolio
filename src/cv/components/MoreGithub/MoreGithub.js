import { isProjectsSection } from "app/slices/animationsSlice";
import Rect from "cv/icons/Rect";
import Github from "svgIcons/Github";
import './MoreGithub.scss';
import { useSelector } from 'react-redux';

const MoreGithub = () => {
    const isProfile = useSelector(isProjectsSection);
   
    return(
        <div className="more-github">
            <div className='content'>
                <Rect height='89' width='328'/>
                <a tabIndex={isProfile ? '-1' : undefined} href="https://github.com/ivaylo9512">
                    <span>More on</span>
                    <Github />
                </a>
            </div>

        </div>
    )
}

export default MoreGithub