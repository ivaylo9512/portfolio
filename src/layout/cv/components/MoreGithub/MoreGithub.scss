.more-github{
    width: 46%;
    box-sizing: border-box;
    position: relative;
    font-family: 'Myriad Pro Regular';
    margin: auto;
    font-size: 3.2vw;


    &:hover,
    &:focus{
        transform: scale(1.1);
    }
    
    .content{
        position: relative;

        a{
            position: absolute;
            height: 100%;
            width: 100%;
            top: 0;
            display: flex;
            justify-content: center;
            align-items: center;
            text-decoration: none;
            color: black;
            text-shadow: 0 0 0.1vw;
            font-weight: bold;
            padding: 0 5%;
            box-sizing: border-box;

            .github{
                width: 13%;
                margin-left: 5%;
            }
        }

        rect{
            fill: white;
            filter: url(#shape-shadow)
        }
    }

    @media only screen and (min-width: 700px) and (orientation: landscape) {
        font-size: 1.2vw;
        padding: 0 9%;
        margin: 0;
    }
}