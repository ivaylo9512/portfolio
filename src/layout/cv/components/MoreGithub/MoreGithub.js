import Rect from '@svgIcons/Rect';
import Github from '@svgIcons/Github';
import './MoreGithub.scss';
import TabLink from '@cv/components/TabLink/TabLink';

const MoreGithub = () => {
    return(
        <div className='more-github'>
            <div className='content'>
                <Rect height='89' width='328'/>
                <TabLink shouldMatch={true} href={'https://github.com/ivaylo9512'}> 
                    <span>More on</span>
                    <Github />
                </TabLink>
            </div>

        </div>
    )
}

export default MoreGithub