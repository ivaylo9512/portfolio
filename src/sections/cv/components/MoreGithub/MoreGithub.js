import { isProjectsSection } from 'app/slices/animationsSlice';
import Rect from '@svgIcons/Rect';
import Github from '@svgIcons/Github';
import './MoreGithub.scss';
import { useSelector } from 'react-redux';

const MoreGithub = ({ shouldEqual }) => {
    const isProjects = useSelector(isProjectsSection);
   
    return(
        <div className='more-github'>
            <div className='content'>
                <Rect height='89' width='328'/>
                <a data-testid='github-link' tabIndex={isProjects == shouldEqual ? '-1' : undefined} href='https://github.com/ivaylo9512'>
                    <span>More on</span>
                    <Github />
                </a>
            </div>

        </div>
    )
}

export default MoreGithub