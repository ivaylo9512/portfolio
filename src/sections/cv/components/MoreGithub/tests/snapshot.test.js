import { shallow } from 'enzyme';
import MoreGithub from '../MoreGithub';
import * as Redux from 'react-redux';

describe('MoreGithub snapshot tests', () => {
    let selectorSpy;
   
    beforeEach(() => {
        selectorSpy = jest.spyOn(Redux, 'useSelector');
    })

    const createWrapper = (state, shouldEqual) => {
        selectorSpy.mockReturnValue(state);
        
        return shallow(
            <MoreGithub shouldEqual={shouldEqual}/>
        )
    }

    it('should match snapshot with matching section', () => {
        const wrapper = createWrapper(true, true);

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot without matching section', () => {
        const wrapper = createWrapper(true, false);

        expect(wrapper).toMatchSnapshot();
    })
})