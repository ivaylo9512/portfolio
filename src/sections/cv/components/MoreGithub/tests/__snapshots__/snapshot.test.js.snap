// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreGithub snapshot tests should match snapshot on matching section 1`] = `
<div
  className="more-github"
>
  <div
    className="content"
  >
    <Rect
      height="89"
      width="328"
    />
    <a
      data-testid="github-link"
      href="https://github.com/ivaylo9512"
      tabIndex="-1"
    >
      <span>
        More on
      </span>
      <Github />
    </a>
  </div>
</div>
`;

exports[`MoreGithub snapshot tests should match snapshot on not matching section 1`] = `
<div
  className="more-github"
>
  <div
    className="content"
  >
    <Rect
      height="89"
      width="328"
    />
    <a
      data-testid="github-link"
      href="https://github.com/ivaylo9512"
    >
      <span>
        More on
      </span>
      <Github />
    </a>
  </div>
</div>
`;
